pdns/receiver.cc

```diff
@@ -605,8 +605,24 @@ int main(int argc, char **argv)
     }
 
     UeberBackend::go();
-    N=new UDPNameserver; // this fails when we are not root, throws exception
-    
+    N=std::make_shared<UDPNameserver>(); // this fails when we are not root, throws exception
```

Comment:
It will still fail if we try to bind to a privileged port and don't have the relevant capabilities. This might happen if we are not started as root, for example.